### ISSUE TITLE
Bug in tf.keras.layers.ReLU.__init__

### ISSUE THREAD

**Issue by blaueck** [tensorflow/tensorflow#22687]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
windows 10
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
source
- **TensorFlow version (use command below)**:
r1.11
- **Python version**:
3.6
- **Bazel version (if compiling from source)**:
0.15.0
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
There is a bug in tf.keras.layers.ReLU.\_\_init__. When passing keyward argument max_value with value None, which is also the default value, no gradient will be backword through it. I think is caused by the following code in the \_\_init__ function:
``` python
self.max_value = K.cast_to_floatx(max_value)
```
When max_value is None, the self.max_value will be nan of type float32. The self.max_value will be used with tf.clip_by_value as the upper bound and cause no gradient, since x < nan is false for any x.  I think the self.max_value should be None when max_value is None. 
``` python
self.max_value = K.cast_to_floatx(max_value) if max_value is not None else None
```

**Comment by wt-huang** [comment]
@blaueck Actually this is not a bug. When max_value is set as default value None, it means that no clipping is applied. Gradient will be either 0 or 1 depending on whether x is positive or negative. There is a zero gradient issue that can cause the output to become zero. This is addressed by modified ReLU such as ELU or leaky ReLU.

**Comment by blaueck** [comment]
Hi @wt-huang , thanks for your reply. What I means is that tf.keras.layers.ReLU don't behave as you say. When I set max_value as a not None value, such as 5, every thing is normal. But when I set max_value as None, there is not gradient backward thought the layer. 
I took a look at the code and found the max_value may be the problem. tf.keras.layers.ReLU actually call tf.keras.backend.relu to do the computation. The following code show how it works (with some simplification).
```python
# -- code in tf.keras.layers.ReLU.__init__ --
# The same as self.max_value = np.asarray(max_value, dtype='float32').
# When max_value is None, self.max_value will equal NaN
self.max_value = K.cast_to_floatx(max_value)


# -- code in tf.keras.backend.relu --
# passing self.max_value to tf.keras.backend.relu as max_value

x = tf.nn.relu(x)

# Since max_value is not None, the condition will eval to True
if max_value is not None:
    x = tf.clip_by_value(x, 0, max_value)
    # When max_value is NaN: x = tf.clip_by_value(x, 0, NaN).
    # This clip op cause gradient to be zero, because no number is between
    # 0 and NaN?
```
Code to reproduce the bug:
``` python
import tensorflow as tf
tf.enable_eager_execution()
# a = [-0.07873772,  1.2001798 ,  0.27422413,  0.8521496 ,  1.0448941 ]
a = tf.random_normal([5])

with tf.GradientTape() as tape:
    tape.watch(a)
    # b = [0.        , 1.2001798 , 0.27422413, 0.8521496 , 1.0448941 ]
    b = tf.keras.layers.ReLU()(a)
# da1 = [0., 0., 0., 0., 0.]
da1 = tape.gradient(b, a)

with tf.GradientTape() as tape:
    tape.watch(a)
    # c = [0.        , 1.2001798 , 0.27422413, 0.8521496 , 1.0448941 ]
    c = tf.keras.layers.ReLU(max_value=5)(a)
# da2 = [0., 1., 1., 1., 1.]
da2 = tape.gradient(c, a)
```

**Comment by daavoo** [comment]
@blaueck Could you point to the commit where it has been fixed??

**Comment by blaueck** [comment]
I think the commit is 1d1ec99bd3b322ea35a2d3d0eb754589ec2fd512.
